js/map.js

```diff
@@ -3,6 +3,36 @@ function htmlentities(str) {
   return String(str).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;');
 }
 $(document).ready(function() {
+  // main map cache
+  var cache = {};
+
+    window.addEventListener("message", function (evt) {
+      var data = JSON.parse(evt.data);
+      switch(data.cmd){
+        case 'catch_alert':
+          overpass.handlers["onAjaxError"] = function(errmsg) {
+              parent.postMessage(JSON.stringify({handler: "onAjaxError",
+                                                 msg: errmsg }), "*");
+          };
+          overpass.handlers["onQueryError"] = function(errmsg) {
+              parent.postMessage(JSON.stringify({handler: "onQueryError",
+                                                 msg: errmsg }), "*");
+          };
+          break;
+        case 'update_map':
+          settings.code["overpass"] = data.value[0];
+          ide.update_map();
+          break;
+        case 'run_query':
```

Comment:
this could use a better name (to make sure it's immediately associated as the caching endpoint)